doc/devel/contributing.rst

```diff
@@ -285,30 +285,57 @@ Coding guidelines
 API changes
 -----------
 
-Changes to the public API must follow a standard deprecation procedure to
-prevent unexpected breaking of code that uses Matplotlib.
-
-- Deprecations must be announced via a new file in
-  a new file in :file:`doc/api/next_api_changes/deprecations/` with
-  naming convention ``99999-ABC.rst`` where ``99999`` is the pull request
-  number and ``ABC`` are the contributor's initials.
-- Deprecations are targeted at the next point-release (i.e. 3.x.0).
-- The deprecated API should, to the maximum extent possible, remain fully
-  functional during the deprecation period. In cases where this is not
-  possible, the deprecation must never make a given piece of code do something
-  different than it was before; at least an exception should be raised.
```

Comment:
I think this point ("must never make a given piece of code...") is important and must be kept (possibly reworded).

Reply:
@jklymak @timhoffm Do you feel the same way regarding this statement? I think @timhoffm consciously changed that part to make it clear that the old API must (not should) remain functional. Therefore, any ",but..." clause is irrelevant. BTW, this is not my definite opinion, I am just paraphrasing what I think @timhoffm said and I can follow this logic. If it MUST stay functional than we don't need to worry about the "what if it does'nt remain functional" case

Reply:
I'm fine with the more stringent statement to keep things simple and as a guide to contributors.  

Very occasionally, this list of rules ends up being bypassed.  Since this basically has to be done by the lead devs, listing the caveats and how to mitigate them seems unnecessary in a document like this.  However, if I were writing the first sentence of the second paragraph I would say 

`API changes in Matplotlib have to be performed following the deprecation process below, except in very rare circumstances as deemed necessary by the development team`.  

That makes it clear we can't always live up to the ideals of this process...

Reply:
I'm OK with @jklymak's take on this.

Reply:
Alright, sounds like a good compromise